src/core/dyad_core.h

```diff
@@ -0,0 +1,192 @@
+#ifndef __DYAD_CORE_CLIENT_H__
+#define __DYAD_CORE_CLIENT_H__
+
+// Includes <flux/core.h>
+#include "dyad_flux_log.h"
+#include "dyad_err.h"
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstdio>
+#else
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#endif
+
+/*****************************************************************************
+ *                                                                           *
+ *                          DYAD Macro Definitions                           *
+ *                                                                           *
+ *****************************************************************************/
+
+#define DYAD_PATH_PROD_ENV "DYAD_PATH_PRODUCER"
+#define DYAD_PATH_CONS_ENV "DYAD_PATH_CONSUMER"
+#define DYAD_CHECK_ENV     "DYAD_SYNC_HEALTH"
+// Now defined in src/utils/utils.h
+//#define DYAD_PATH_DELIM    "/"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * @struct dyad_ctx
+ */
+struct dyad_ctx {
+    flux_t *h; // the Flux handle for DYAD
+    bool debug; // if true, perform debug logging
+    bool check; // if true, perform some check logging
+    bool reenter; // if false, do not recursively enter DYAD
+    bool initialized; // if true, DYAD is initialized
+    bool shared_storage; // if true, the managed path is shared
+    bool sync_started; // TODO
+    unsigned int key_depth; // Depth of bins for the Flux KVS
+    unsigned int key_bins; // Number of bins for the Flux KVS
+    uint32_t rank; // Flux rank for DYAD
+    char *kvs_namespace; // Flux KVS namespace for DYAD
+    char *prod_managed_path; // producer path managed by DYAD
+    char *cons_managed_path; // consumer path managed by DYAD
+    bool intercept; // if true, intercepts (f)open and (f)close
```

Comment:
I think reenter was serving the same purpose in the original code, but I might be wrong.

Reply:
Similar to the `dyad_open_dlwrap.h` stuff, the `intercept` flag has been phased out in favor of `reenter`. I'll remove it.